beacon-chain/rpc/service_test.go

```diff
@@ -113,8 +117,33 @@ func (m mockChainService) IsCanonical(slot uint64, hash []byte) bool {
 	return bytes.Equal(m.canonicalBlocks[slot], hash)
 }
 
-func (m mockChainService) RecentCanonicalRoots(count uint64) []*pbrpc.BlockRoot {
-	return nil
+func (m *mockChainService) BlockChildren(ctx context.Context, block *pb.BeaconBlock, highestSlot uint64) ([]*pb.BeaconBlock, error) {
+	blockRoot, err := hashutil.HashBeaconBlock(block)
+	if err != nil {
+		return nil, err
+	}
+	var children []*pb.BeaconBlock
+	startSlot := block.Slot + 1
+	for i := startSlot; i <= highestSlot; i++ {
+		kids, err := m.beaconDB.BlocksBySlot(ctx, i)
```

Comment:
This loop is the expensive thing i was talking about offline. We should check that ctx is not expired here. Maybe even can be within BlocksBySlot